File: schoollicensemanager/src/main/java/br/com/totvs/schoollicensemanager/service/TeacherService.java
package br.com.totvs.schoollicensemanager.service;

import org.springframework.data.domain.Page;

import br.com.totvs.schoollicensemanager.model.Teacher;

public interface TeacherService {

	public Page<Teacher> findAll(String page, String size);

	public Teacher findById(Long id);

	public Teacher create(Teacher teacher);

	public Teacher update(Teacher teacher);
	
	public Page<Teacher> findByNameContaining(String name, String page, String size);
	
	public void delete(Long id);

}

File: schoolLicenseManagerThf/src/app/core/teacher.service.ts
import { HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { PageResponseEntity } from '../model/pageResponseEntity';
import { Teacher } from '../model/teacher';
import { CrudService } from './crud.service';

export const TEACHER_URI = `teachers`;
@Injectable({
  providedIn: 'root'
})
export class TeacherService extends CrudService<Teacher> {
  initializeEndpoint(): void {
    this.endpoint = TEACHER_URI;
  }

  findByNameContaining(name: string, page: string, size: string): Observable<PageResponseEntity<Teacher>> {
    const params = new HttpParams().set('name', name).set('page', page).set('size', size);
    return this.http.get<PageResponseEntity<Teacher>>(`${this.getUri()}`, { params: params });
  }
}
